addons/account/models/account_move.py

```diff
@@ -3750,6 +3750,17 @@ def invalidate_cache(self, fnames=None, ids=None):
                 self.env['account.move'].invalidate_cache(ids=move_ids)
         return super().invalidate_cache(fnames=fnames, ids=ids)
 
+    @api.model
+    def _name_search(self, name, args=None, operator='ilike', limit=100, name_get_uid=None):
+        if operator == 'ilike':
+            args = ['|', '|',
+                    ('name', '=', name),
+                    ('move_id.name', '=', name),
+                    ('product_id.display_name', '=', name)]
+            # remove name, so parent _name_search will not add it to the domain again
+            name = ''
```

Comment:
Usually, one would do a `_search` afterwards

Reply:
If you are changing the input anyways and basically disabling all future calls to `super`, I don't see the point...